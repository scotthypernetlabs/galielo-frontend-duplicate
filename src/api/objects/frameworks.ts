import {
  convertToFrameworkOptions,
  convertToFrameworkSteps,
  convertToFrameworkWidget
} from "../../data/implementations/frameworkRepository";

export interface IFrameworkReceived {
  name: string;
  id: string;
  versions: string[];
}

export interface IGetFrameworks {
  frameworks: IFrameworkReceived[];
}

export interface IFrameworkOptions {
  name: string;
  value: string;
  hover: string;
  description: string;
  widgets: IFrameworkWidget[];
}

export interface IFrameworkWidget {
  type: string;
  header: string;
  subheader?: string;
  key: string;
  default?: boolean | number;
  options?: IFrameworkOptions[];
  filter?: string;
  placeholder?: string;
  mask?: string;
  required?: boolean;
  widgets: IFrameworkWidget[];
}

export interface IFrameworkStep {
  page: number;
  header: string;
  subheader: string;
  widgets: IFrameworkWidget[];
}

export interface IFrameworkExpanded {
  id: string;
  name: string;
  description: string;
  steps: IFrameworkStep[];
}

export class FrameworkExpanded {
  public steps: FrameworkSteps[];
  constructor(
    public id: string,
    public name: string,
    public description: string,
    s: IFrameworkStep[]
  ) {
    this.steps = [];
    s.forEach((step: IFrameworkStep) => {
      this.steps.push(convertToFrameworkSteps(step));
    });
  }
}

export class FrameworkOptions {
  public widgets: FrameworkWidget[];
  constructor(
    public name: string,
    public value: string,
    public hover: string,
    public description: string,
    w?: IFrameworkWidget[]
  ) {
    this.widgets = [];
    w.forEach((widget: IFrameworkWidget) => {
      this.widgets.push(convertToFrameworkWidget(widget));
    });
  }
}

export class FrameworkWidget {
  public widgets: FrameworkWidget[];
  public options: FrameworkOptions[];
  constructor(
    public type: string,
    public header: string,
    public key: string,
    public subheader?: string,
    public defaultValue?: boolean | number,
    o?: IFrameworkOptions[],
    public filter?: string,
    public placeholder?: string,
    public mask?: string,
    public required?: boolean,
    w?: IFrameworkWidget[]
  ) {
    this.widgets = [];
    this.options = [];
    w.forEach((widget: IFrameworkWidget) => {
      this.widgets.push(convertToFrameworkWidget(widget));
    });
    o.forEach((options: IFrameworkOptions) => {
      this.options.push(convertToFrameworkOptions(options));
    });
  }
}

export class FrameworkSteps {
  public widgets: FrameworkWidget[];
  constructor(
    public page: number,
    public header: string,
    public subheader: string,
    w: IFrameworkWidget[]
  ) {
    this.widgets = [];
    w.forEach((widget: IFrameworkWidget) => {
      this.widgets.push(convertToFrameworkWidget(widget));
    });
  }
}

export class FrameworkReceived {
  constructor(name: string, id: string, versions: string[]) {}
}
