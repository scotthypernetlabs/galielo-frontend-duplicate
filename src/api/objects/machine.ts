export interface IMachine {
  name: string;
  userid: string;
  status: string;
  id: string;
  gpu: string;
  cpu: string;
  os: string;
  arch: string;
  memory: string; 
  jobs_in_queue: number;
  running_jobs_limit: number;
  running_jobs: number;
}
