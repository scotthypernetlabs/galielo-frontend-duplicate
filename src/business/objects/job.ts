import { Dictionary } from './dictionary';
import store from '../../store/store';
import { updateStationUploadProgress, updateMachineUploadProgress } from '../../actions/progressActions';
// import DateTimeFormat = Intl.DateTimeFormat;

export class Job {
  constructor(
    public container: string,
    public id: string,
    public last_updated: number, // timestamp of last entry in status history
    public name: string,
    public oaid: string,
    public pay_interval: number,
    public pay_status: EPaymentStatus,
    public landing_zone: string,
    public launch_pad: string,
    public job_state: EJobRunningStatus,
    public station_id: string,
    public status: EJobStatus,
    public status_history: JobStatus[],
    public upload_time: number,
    public run_time: number,
  ) {}
}

export class JobStatus {
  constructor(
    public jobid: string,
    public jobstatusid: string,
    public status: EJobStatus,
    public timestamp: number
  ){

  }
}

export enum EJobStatus {
  uploaded = "uploaded",
  submitted = "submitted",
  downloaded = "downloaded",
  building_image = "building_image",
  building_container = "building_container",
  start_requested = "start_requested",
  running = "running",
  pause_requested = "pause_requested",
  paused = "paused",
  stop_requested = "stop_requested",
  stopped = "stopped",
  exited = "exited",
  collecting_results = "collecting_results",
  posting_results = "posting_results",
  results_posted = "results_posted",
  terminated = "terminated",
  completed = "completed",
  removed_by_host = "removed_by_host",
  unknown = "unknown",
  post_processing = "post_processing",
  started = "started"
}

export enum EJobRunningStatus {
  not_running = "not_running",
  running = "running"
}

export enum EPaymentStatus {
  current = "current",
  payment_due = "payment_due",
  delinquent = "delinquent",
  missing_offer = "missing_offer"
}

export class GetJobFilters {
  constructor(
    public jobids?: string[],
    public receiverids?: string[],
    public userids?: string[],
    public stationids?: string[],
    public statuses?: string[],
    public page?: number,
    public items?: number
  ){

  }
}

export class DockerLog {
  constructor(
    public Processes: string[],
    public Titles: string[]
  ){

  }
}

export class UploadQueue {
  public queue: Function[] = [];
  public totalQueued: number = 0;
  public totalFinished: number = 0;
  public running: boolean = false;
  public componentsToUpdate: Dictionary<React.Component> = {};
  constructor(
  ){
  }
  // Components that need to re-render based on changes in the Queue
  bindComponent(component: React.Component, identity: string){
    this.componentsToUpdate[identity] = component;
  }
  // Function to call the re-render on the components that need to be updated.
  updateComponents(finalUpload?: boolean){
    Object.keys(this.componentsToUpdate).forEach((identity: string) => {
      if(finalUpload){
        this.componentsToUpdate[identity].setState({fileUploadText:"Browse or drop directory"})
      }else{
        this.componentsToUpdate[identity].forceUpdate();
      }
    })
  }
  // Component that was listening for updates to the queue will unmount
  removeComponent(identity: string){
    delete this.componentsToUpdate[identity]
  }
  addToQueue(callback: Function){
    this.queue.push(callback);
    this.totalQueued += 1;
    this.updateComponents();
  }
  startQueue(){
    if(!this.running){
      this.running = true;
      this.startNext();
    }
    window.onbeforeunload = function(){ return true }
  }
  async startNext(){
    if(this.length() > 0){
      let next = this.queue.shift();
      await next();
      this.totalFinished += 1;
      this.startNext();
      this.updateComponents();
    }else{
      setTimeout(() => {
        this.totalQueued = 0;
        this.totalFinished = 0;
        this.running = false;
        this.updateComponents(true);
        window.onbeforeunload = null;
      }, 3000);
    }
  }
  length(){
    return this.queue.length;
  }
}

export class UploadObjectContainer {
  constructor(
    public project_id: string,
    public uploadingFiles: UploadObject[],
    public totalUploadSize: number,
    public completedUploadSize: number,
    public station_id?: string,
    public machine_id?: string
  ){
  }
  addUploadingFile(uploadObject: UploadObject){
    this.uploadingFiles.push(uploadObject);
    this.totalUploadSize += uploadObject.total;
  }
  updateProgress(uploadObject: UploadObject){
    this.completedUploadSize = this.completedUploadSize + uploadObject.loaded - uploadObject.previousLoaded;
    uploadObject.previousLoaded = uploadObject.loaded;
    if(this.station_id){
      store.dispatch(updateStationUploadProgress(this));
    }
    if(this.machine_id){
      store.dispatch(updateMachineUploadProgress(this));
    }
  }
  addEventListeners(type: EventListenerTypes, eventListener: Function){
    this.uploadingFiles.forEach((uploadObject: UploadObject) => {
      uploadObject.xhrObject.upload.addEventListener(type, (e: ProgressEvent) => eventListener(e));
    })
  }
  cancelAllUploads(){
    this.uploadingFiles.forEach((uploadObject: UploadObject) => {
      console.log("Aborting");
      uploadObject.xhrObject.abort();
    })
  }
}

export class UploadObject {
  constructor(
    public xhrObject: XMLHttpRequest,
    public previousLoaded: number,
    public loaded: number,
    public total: number
  ){
  }

}

export enum EventListenerTypes {
  abort = "abort",
  error = "error",
  load = "load",
  loadend = "loadend",
  loadstart = "loadstart",
  progress = "progress",
  timeout = "timeout"
}
type MyMapLikeType = Record<string, string>;

export const JobStatusDecode: MyMapLikeType = {
  uploaded: "Queued",
  submitted: "Job Uploading",
  downloaded: "Job Uploading",
  building_image: "Building Image",
  built_image: "Building Image",
  building_container: "Building Container",
  built_container: "Built Container",
  start_requested: "Job In Progress",
  started: "Job In Progress",
  running: "Job In Progress",
  pause_requested: "Job Paused",
  paused: "Job Paused",
  stop_requested: "Job Cancelled",
  stopped: "Job Cancelled",
  terminated: "Job Cancelled",
  collecting_results: "Collecting Results",
  post_processing: "Collecting Results",
  posting_results: "Collecting Results",
  results_posted: "Results Received",
  exit_error: "Exit Error",
  completed: "Completed",
  build_error: "Build Error",
  docker_error: "Docker Error",
  unknown: "Error",
  queued: "Queued"
};

// export function getEnumKeyByEnumValue<T>(myEnum: any, enumValue: string|number):T {
//     let keys = Object.keys(myEnum).filter(x => myEnum[x] == enumValue);
//     return keys.length > 0 ? keys[0] : null;
// }
