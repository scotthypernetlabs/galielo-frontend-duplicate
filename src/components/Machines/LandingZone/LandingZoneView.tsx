import {
  AppBar,
  Box,
  Button,
  Dialog,
  DialogActions,
  DialogContent,
  DialogContentText,
  DialogTitle,
  FormControl,
  Grid,
  IconButton,
  TextField,
  Toolbar,
  Typography
} from "@material-ui/core";
import {
  faCircle,
  faSdCard,
  faTachometerAlt
} from "@fortawesome/free-solid-svg-icons";
import { green, red } from "../../theme";
import IconText from "../../Core/IconText";
import ProgressBar from "../../ProgressBar";
import React, { useState } from "react";

interface LandingZoneViewProps {
  machineStatus: string;
  machineName: string;
  machineId: string;
  memoryText: string;
  coresText: string;
  uploadText: string;
  inStation: boolean;
  showText: boolean;
  updateRunningJobLimit: any;
  machineOwner: string;
  machineCpu: string;
  machineOS: string;
  machineArch: string;
  machineJobsInQueue: number;
  machineRunningJobsLimit: number;
  machineRunninJobs: number;
}

const LandingZoneView: React.SFC<LandingZoneViewProps> = (
  props: LandingZoneViewProps
) => {
  const {
    machineStatus,
    machineName,
    machineId,
    memoryText,
    coresText,
    uploadText,
    inStation,
    showText,
    updateRunningJobLimit,
    machineOwner,
    machineCpu,
    machineOS,
    machineArch,
    machineJobsInQueue,
    machineRunningJobsLimit,
    machineRunninJobs
  } = props;
  const [open, setOpen] = useState(false);
  const [runningJobsLimit, setRunningJobsLimit] = useState(0);
  const [changed, setChanged] = useState(false);
  const handleClickOpen = (e: any) => {
    if (!open && !inStation) {
      setOpen(true);
    }
  };
  const handleClose = (e: any) => {
    setOpen(false);
  };
  const handleRunningJobsChange = (e: any) => {
    setRunningJobsLimit(e.target.value);
  };

  const submit = () => {
    updateRunningJobLimit(machineId, runningJobsLimit);
    setOpen(false);
  };
  const limitChanged = (e: any) => {
    if (e.target.value > 0) {
      setChanged(true);
    } else if (e.target.value == 0) {
      setChanged(false);
    }
  };

  const boxClasses = () => {
    return [inStation ? "station-box" : "", "button"].join(" ");
  };
  return (
    <Box
      onClick={handleClickOpen}
      border={inStation ? "2px dashed" : "0.5px solid"}
      borderColor={
        inStation
          ? machineStatus === "ONLINE"
            ? "primary.main"
            : "red"
          : "#999"
      }
      borderRadius={5}
      bgcolor="rgb(255, 255, 255, 0.5)"
      p={3}
      m={1}
      minWidth="250px"
      maxWidth="250px"
      minHeight={inStation ? "130px" : "100px"}
      maxHeight={inStation ? "130px" : "100px"}
      className={boxClasses()}
    >
      <Box display="flex" flexWrap="noWrap" mb={0.75} mr={4}>
        <IconText
          icon="fiber_manual_record"
          text={machineName}
          textVariant="h4"
          noWrap={true}
          iconColor={machineStatus === "ONLINE" ? green.main : red.main}
          iconSize={14}
        />
      </Box>
      <Box
        display="flex"
        flexDirection="row"
        justifyContent="flex-start"
        mb={0.75}
        mr={4}
      >
        <Box flexGrow={1}>
          <IconText
            icon="sd_card"
            text={memoryText}
            textVariant="h5"
            textColor="grey"
            iconSize={16}
          />
        </Box>
        <Box>
          <IconText
            icon="speed"
            text={coresText}
            textVariant="h5"
            textColor="grey"
            iconSize={18}
          />
        </Box>
      </Box>
      {showText && (
        <Box mb={0.75}>
          <Typography variant="h5">{uploadText}</Typography>
        </Box>
      )}
      <Box mb={0.75}>
        <ProgressBar type={"machine"} id={machineId} />
      </Box>
      <Dialog
        // fullScreen
        fullWidth={true}
        maxWidth={"md"}
        open={open}
        aria-labelledby="alert-dialog-title"
        aria-describedby="alert-dialog-description"
      >
        <Box p={3}>
          <Typography id="alert-dialog-title">{machineName}</Typography>
          <Grid container spacing={3}>
            <Grid item xs={6}>
              <Box mb={1} mt={1}>
                <Typography variant="h6" id="alert-dialog-title">
                  Status
                </Typography>
                <Typography variant="h5" id="alert-dialog-title">
                  {machineStatus}
                </Typography>
              </Box>
              <Box mb={1}>
                <Typography variant="h6" id="alert-dialog-title">
                  Cores
                </Typography>
                <Typography variant="h5" id="alert-dialog-title">
                  {coresText}
                </Typography>
              </Box>

              <Box mb={1}>
                <Typography variant="h6" id="alert-dialog-title">
                  Memory
                </Typography>
                <Typography variant="h5" id="alert-dialog-title">
                  {memoryText}
                </Typography>
              </Box>
            </Grid>
            <Grid item xs={6}>
              <Box mb={1} mt={1}>
                <Typography variant="h6" id="alert-dialog-title">
                  CPU
                </Typography>
                <Typography variant="h5" id="alert-dialog-title">
                  {machineCpu}
                </Typography>
              </Box>
              <Box mb={1}>
                <Typography variant="h6" id="alert-dialog-title">
                  OS
                </Typography>
                <Typography variant="h5" id="alert-dialog-title">
                  {machineOS}
                </Typography>
              </Box>
              <Box mb={1}>
                <Typography variant="h6" id="alert-dialog-title">
                  Job Limits
                </Typography>
                <Typography variant="h5" id="alert-dialog-title">
                  {machineRunningJobsLimit}
                </Typography>
              </Box>
            </Grid>
          </Grid>

          <Box mt={1}>
            <DialogContentText id="alert-dialog-description">
              Update running jobs limit:
            </DialogContentText>
            <TextField
              autoFocus
              onBlur={handleRunningJobsChange}
              onChange={limitChanged}
              margin="dense"
              id="limit"
              InputProps={{ inputProps: { min: 0, max: 10 } }}
              label="Enter a value between 1 and 10"
              type="number"
              variant="outlined"
            />
          </Box>
          <DialogActions>
            <Button onClick={handleClose} color="primary">
              Cancel
            </Button>
            <Button
              disabled={!changed}
              onClick={submit}
              color="primary"
              autoFocus
            >
              Submit
            </Button>
          </DialogActions>
        </Box>
      </Dialog>
    </Box>
  );
};

export default LandingZoneView;
