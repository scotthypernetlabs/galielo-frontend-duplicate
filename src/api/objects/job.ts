import { EPaymentStatus, EJobRunningStatus, EJobStatus } from "../../business/objects/job";
// import DateTimeFormat = Intl.DateTimeFormat;

export interface IJob {
  container: string;
  jobid: string;
  last_updated: number;
  name: string;
  oaid: string; // offer accept id
  pay_interval: number;
  pay_status: EPaymentStatus;
  receiverid: string;
  userid: string; // sender of job
  state: EJobRunningStatus;
  stationid: string;
  status: EJobStatus;
  status_history: IJobStatusHistory[];
  time_created: number;
  total_runtime: number;
  senderid: string; // machine right now
  archived: boolean;
}

export interface IJobStatusHistory {
  jobid: string;
  jobstatusid: string;
  status: EJobStatus;
  timestamp: number;
}
