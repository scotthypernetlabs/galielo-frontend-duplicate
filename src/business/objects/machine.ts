import { Dictionary } from "./dictionary";
import { IMachine } from "../../api/objects/machine";

export class Machine {
  constructor(
    public machine_name: string,
    public owner: string,
    public status: string,
    public mid: string,
    public gpu: string,
    public cpu: string,
    public os: string,
    public arch: string,
    public memory: string,
    public jobs_in_queue: number,
    public running_jobs_limit: number,
    public running_jobs: number
  ) {}
}

export class GetMachinesFilter {
  constructor(public mids?: string[], public userids?: string[]) {}
}

export function convertToBusinessMachine(machines: IMachine[]) {
  return machines.map(machine => {
    return new Machine(
      machine.name,
      machine.userid,
      machine.status,
      machine.mid,
      machine.gpu,
      machine.cpu,
      machine.os,
      machine.arch,
      machine.memory,
      machine.jobs_in_queue,
      machine.running_jobs_limit,
      machine.running_jobs
    );
  });
}
