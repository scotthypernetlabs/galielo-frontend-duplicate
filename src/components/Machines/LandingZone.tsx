import React from 'react';
import { Machine } from '../../business/objects/machine';
import {Box, Grid, Typography} from "@material-ui/core";
import {FontAwesomeIcon} from "@fortawesome/react-fontawesome";
import {faSdCard, faTachometerAlt} from "@fortawesome/free-solid-svg-icons";
import ProgressBar from "../ProgressBar";

type Props = {
  machine: Machine;
  station: boolean;
  fileUploadText?: string;
}

type State = {

}

class LandingZone extends React.Component<Props, State> {
  constructor(props: Props){
    super(props);
  }
  public render(){
    const { machine } = this.props;
    let memory:number = 0;
    let cores:number = 0;
    if(machine.memory !== 'Unknown'){
      memory = parseInt((parseInt(machine.memory) / 1e9).toFixed(1));
    }
    const iconColor = machine.status.toUpperCase() === 'ONLINE' ? "rgb(40, 202, 66)" : "red";

    return(
      <Box
        border="2px dashed"
        borderColor="primary.main"
        borderRadius={5}
        bgcolor="white"
        p={3}
        m={1}
        minWidth="250px"
        maxWidth="250px"
        minHeight="100px"
        maxHeight="100px"
      >
        <Grid container>
          <Grid item xs={12}>
            <svg width="10px" height="10px" viewBox="0 0 100 100" version="1.1" style={{float: "left", margin: "5px 5px 5px 0px"}}>
              <ellipse id="online-badge" fill={iconColor} cx="50" cy="50" rx="40" ry="40" />
            </svg>
            <Typography variant="h4" noWrap={true} gutterBottom={true} style={{fontWeight: 500}}>{machine.machine_name}</Typography>
          </Grid>
          <Grid item xs={6}>
            <FontAwesomeIcon icon={faSdCard} color="grey" style={{float: 'left', marginRight: 5}}/>
            <h5 style={{color: "grey", fontWeight: 400}}>{memory}GB</h5>
          </Grid>
          <Grid item xs={6}>
            <FontAwesomeIcon icon={faTachometerAlt} color="grey" style={{float: 'left', marginRight: 5}}/>
            <h5 style={{color: "grey", fontWeight: 400}}>{cores} Cores</h5>
          </Grid>
          {
            this.props.station &&
              <div>
                <Grid item xs={12}>
                    <h5>{this.props.fileUploadText}</h5>
                </Grid>
              </div>
          }
          <Grid item xs={12}>
            <ProgressBar mid={machine.mid}/>
          </Grid>
        </Grid>
      </Box>
    )
  }
}

export default LandingZone;
