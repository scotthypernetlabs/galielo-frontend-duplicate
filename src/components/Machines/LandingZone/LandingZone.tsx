import { Dictionary } from "../../../business/objects/dictionary";
import { IStore } from "../../../business/objects/store";
import { Machine } from "../../../business/objects/machine";
import { MyContext } from "../../../MyContext";
import { UploadObjectContainer } from "../../../business/objects/job";
import { connect } from "react-redux";
import { context } from "../../../context";
import LandingZoneView from "./LandingZoneView";
import React from "react";

type Props = {
  machine: Machine;
  station: boolean;
  fileUploadText?: string;
  stationUploads: Dictionary<UploadObjectContainer>;
  machineUploads: Dictionary<UploadObjectContainer>;
};

type State = {
  identity: string;
};

class LandingZone extends React.Component<Props, State> {
  context!: MyContext;
  constructor(props: Props) {
    super(props);
    this.state = {
      identity: "Landing Zone"
    };
    this.updateRunningJobLimit = this.updateRunningJobLimit.bind(this);
  }
  componentDidMount() {
    this.context.uploadQueue.bindComponent(this, this.state.identity);
  }
  componentWillUnmount() {
    this.context.uploadQueue.removeComponent(this.state.identity);
  }
  updateRunningJobLimit(mid: string, runningJobsLimit: number) {
    this.context.machineService.modifyMachineQueueLimit(mid, runningJobsLimit);
  }
  public render() {
    const { machine, station, fileUploadText, machineUploads } = this.props;
    let memory: number = 0;
    let cores: number = 0;
    if (machine.memory !== "Unknown") {
      memory = parseInt((parseInt(machine.memory) / 1e9).toFixed(1));
    }
    if (machine.cpu !== "Unknown") {
      cores = +machine.cpu;
    }
    const memoryText = `${memory}GB`;
    const coresText = `${cores} Cores`;

    let uploadText = fileUploadText;
    const machineUploadProgressObject = machineUploads[machine.mid];
    if (machineUploadProgressObject) {
      const queue = this.context.uploadQueue;
      uploadText = `Uploading Job ${queue.totalFinished + 1} of ${
        queue.totalQueued
      }`;
      const percentage = Math.floor(
        (machineUploadProgressObject.completedUploadSize /
          machineUploadProgressObject.totalUploadSize) *
          100
      );
      if (percentage === 100) {
        setTimeout(() => {
          this.forceUpdate();
        }, 2000);
      }
    }

    return (
      <LandingZoneView
        updateRunningJobLimit = {this.updateRunningJobLimit}
        machineStatus={machine.status.toUpperCase()}
        machineName={machine.machine_name}
        machineId={machine.mid}
        memoryText={memoryText}
        coresText={coresText}
        uploadText={uploadText}
        inStation={station}
        showText={station}
      />
    );
  }
}

LandingZone.contextType = context;

const mapStateToProps = (store: IStore) => ({
  stationUploads: store.progress.stationUploads,
  machineUploads: store.progress.machineUploads
});

export default connect(mapStateToProps)(LandingZone);
