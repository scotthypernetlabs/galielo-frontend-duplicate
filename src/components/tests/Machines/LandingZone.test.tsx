import { Dictionary } from "../../../business/objects/dictionary";
import { Machine } from "../../../business/objects/machine";
import { Provider } from "react-redux";
import { UploadObjectContainer } from "../../../business/objects/job";
import { shallow } from "enzyme";
import LandingZone from "../../Machines/LandingZone/LandingZone";
import React from "react";
import configureMockStore from "redux-mock-store";
import thunk from "redux-thunk";

describe("LandingZone outside of station", () => {
  const stationUploads: Dictionary<UploadObjectContainer> = {};
  const machineUploads: Dictionary<UploadObjectContainer> = {};
  const mockStore = configureMockStore([thunk]);
  const store = mockStore({ progress: { stationUploads, machineUploads } });
  const machine = new Machine(
    "test_machine",
    "hypernet",
    "online",
    "test_mid",
    "GPU",
    "1",
    "MacOS",
    "x64",
    "2",
    3,
    1,
    0
  );
  const landingZone = shallow(
    <Provider store={store}>
      <LandingZone
        machine={machine}
        station={false}
        fileUploadText="Testing..."
      />
    </Provider>
  );

  it("matches snapshot", () => {
    // Snapshot demo
    expect(landingZone).toMatchSnapshot();
  });
});
