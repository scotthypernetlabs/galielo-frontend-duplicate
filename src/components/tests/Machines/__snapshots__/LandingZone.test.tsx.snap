// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LandingZone outside of station matches snapshot 1`] = `ShallowWrapper {}`;
